### PR TITLE
Connects Sign Message dialog with functionality for DApps on Android

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/22547

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Screenshot_20220425-162028_Brave - Debug

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -453,6 +453,14 @@ private void maybeShowAddSuggestTokenRequestLayout() {
         });
     }
 
+    public void dismissWalletPanelOrDialog() {
+        BraveToolbarLayoutImpl layout = (BraveToolbarLayoutImpl) findViewById(R.id.toolbar);
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:457]: nit: can remove explicit cast `(BraveToolbarLayoutImpl)`

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignMessageFragment.java [hunk 2]
```diff
@@ -5,48 +5,177 @@
...
+                        notifySignMessageRequestProcessed(false);
+                    }
+                });
+        view.findViewById(R.id.fragment_sign_msg_btn_sign)
+                .setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        notifySignMessageRequestProcessed(true);
+                    }
+                });
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignMessageFragment.java:114]: nit: use lambda for click listener `(v)->{}`

**Review by Pavneet-Sing — APPROVED** [review]
++

File: android/java/res/layout/fragment_sign_message.xml [hunk 3]
```diff
@@ -4,14 +4,28 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:paddingTop="48dp"
+    android:paddingTop="20dp"
```
Comment (Pavneet-Sing) [inline android/java/res/layout/fragment_sign_message.xml:7]: nit: Could you please use `padding` here as we some need space on all sides.

**Review by Pavneet-Sing — APPROVED** [review]

**Review by qamarngr — APPROVED** [review]
Lgtm

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignMessageFragment.java [hunk 4]
```diff
@@ -5,48 +5,177 @@
...
+    }
+
+    private JsonRpcService getJsonRpcService() {
+        Activity activity = getActivity();
+        if (activity instanceof BraveWalletBaseActivity) {
+            return ((BraveWalletBaseActivity) activity).getJsonRpcService();
+        }
+
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignMessageFragment.java]: it's a good idea, I moved it there

**Pull request merged**